.pre-commit-config.yaml

```diff
@@ -1,8 +1,92 @@
 repos:
-  - repo: https://github.com/psf/black
-    rev: 22.12.0
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v5.0.0
     hooks:
-      - id: black
-        language_version: python3.12
-
-
+      - id: check-added-large-files
+        args: [— maxkb=1000]
+        name: Check for added large files
+      - id: check-ast
+        name: Check Python files for parse errors
+      - id: check-case-conflict
+        name: Check files for case conflicts on case-insensitive filesystems
+      - id: check-docstring-first
+        name: Check that no code is placed prior to docstrings
+      - id: check-executables-have-shebangs
+        name: Check that executables have shebangs
+      - id: check-illegal-windows-names
+        name: Check for illegal Windows filenames
+      - id: check-json
+        name: Check valid JSON syntax
+      - id: check-merge-conflict
+        name: Check for files that contain merge conflict strings
+      - id: check-shebang-scripts-are-executable
+        name: Check that shebang scripts are executable
+      - id: check-toml
+        name: Check valid TOML syntax
+      - id: check-xml
+        name: Check valid XML syntax
+      - id: check-yaml
+        name: Check valid YAML syntax
+      - id: end-of-file-fixer
+        name: Check for newline at end of file
+      - id: debug-statements
+        name: Check for debugger imports
+      - id: no-commit-to-branch
+        name: Check to not commit to main
+      - id: trailing-whitespace
+        name: Check for trailing whitespaces
+  - repo: https://github.com/PyCQA/bandit
+    rev: 1.8.0
+    hooks:
+      - id: bandit
+        name: Run bandit (security linter for python)
+        args: ["-c", "pyproject.toml", "-r", "-ll", "meshpy/"]
+        additional_dependencies: ["bandit[toml]"]
+  - repo: https://github.com/pre-commit/mirrors-clang-format
+    rev: v19.1.5
```

Comment:
Did you try to apply this version to the C++ files in the repo? Maybe we need to "update" them.

Reply:
The C++ files are checked and everything is correct. I've added "unformatted" changes to the .cpp/.h files and they were formatted correctly.

If the files would not adhere to the formatting - the pipeline would fail as it checks all files within the repo